### PR TITLE
Just for better comparison 

### PR DESCRIPTION
**Pull request by tomvoelker**
## Intent of Changes
- Provide safe no-op implementations for search interfaces (NoOpProjectSearch, NoOpPersonSearch, NoOpGroupSearch) so Spring bean wiring remains valid when search is disabled.
- Rewire the Elasticsearch-lite Spring context to alias projectSearch/groupSearch/personSearch to these new no-op beans.
- Add a configuration-driven security.permitDocs flag to SecurityConfig to optionally permit API docs/Swagger endpoints.
- Harden and expand test seed data by converting INSERT -> INSERT IGNORE and adding many group and id rows to reduce duplicate-key failures in tests; update tag visibility and popular_tags seeding.
- Add an npm script "build:docker" and apply small layout/flex-sizing tweaks (md:min-w-0, min-w-0) in the webapp.
- Narrow TypeScript linting scope and relax a few ESLint rules.
- Numerous frontend formatting/refactors, minor UI adjustments, and documentation updates (PROGRESS.md, MVP_API_SWITCHOVER.md, branding/CLAUDE notes, mock data, service worker formatting).

## Main Risks
- Backward compatibility / behavior: No-op search beans return empty lists/zero statistics — callers expecting real search results may see degraded UX or functional regressions.
- Observability/debuggability: Silent fallback to no-op search can make it hard to distinguish “search disabled” from legitimately empty results without logs/metrics.
- Data integrity / test correctness: Using INSERT IGNORE and expanded seed data can mask duplicate-key issues or schema/migration problems and hide failing assumptions in tests.
- Authorization/configuration: security.permitDocs defaults to false; deployments that require docs may be surprised if not documented/enabled.
- Frontend regressions: Multiple Vue edits removed local prop bindings (e.g., PageContainer, EmptyState, TagBadge, PostSection) — templates relying on those bindings may break.
- CI/build: New "build:docker" and ESLint changes require CI validation; formatting and template changes may surface runtime/type errors not caught locally.

## Follow-up Work
- Add logging/metrics or a health indicator to signal when no-op search beans are active (so operators and tests can detect disabled search).
- Add integration and end-to-end tests for critical endpoints and UI flows that depend on search to validate graceful handling of empty results.
- Document security.permitDocs and update deployment configs/README to make enabling API docs explicit.
- Audit Vue components that removed local prop bindings; restore bindings where templates depend on them or add unit tests to verify template prop access.
- Revisit test-seeding strategy: prefer idempotent seeders or migration tests so INSERT IGNORE doesn't mask real issues; add schema/uniqueness checks in CI.
- Validate "build:docker" in CI (containerized build/E2E) and run lints/type-checks after ESLint rule relaxations to detect regressions.

### CODE REVIEW

File: bibsonomy-rest-api-v2/src/main/java/org/bibsonomy/api/search/NoOpGroupSearch.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * Minimal no-op implementation to satisfy legacy beans when search is disabled.
+ */
+public class NoOpGroupSearch implements GroupSearch {
+
+	@Override
+	public List<Group> getGroups(final User loggedinUser, final GroupQuery query) {
+		return Collections.emptyList();
+	}
+}
```
[the remark this answers is not in the document]
Comment (tomvoelker) [reply bibsonomy-rest-api-v2/src/main/java/org/bibsonomy/api/search/NoOpGroupSearch.java]: Resolved: Converted to Kotlin in commit 8ac68e24da.

File: bibsonomy-rest-api-v2/src/main/java/org/bibsonomy/api/search/NoOpPersonSearch.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	@Override
+	public List<Person> getPersons(final PersonQuery query) {
+		return Collections.emptyList();
+	}
+
+	@Override
+	public Statistics getStatistics(final User loggedinUser, final PersonQuery query) {
+		return new Statistics(0);
+	}
+}
```
[the remark this answers is not in the document]
Comment (tomvoelker) [reply bibsonomy-rest-api-v2/src/main/java/org/bibsonomy/api/search/NoOpPersonSearch.java]: Resolved: Converted to Kotlin in commit 8ac68e24da.

File: bibsonomy-rest-api-v2/src/main/java/org/bibsonomy/api/search/NoOpProjectSearch.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	@Override
+	public Statistics getStatistics(final User loggedinUser, final ProjectQuery query) {
+		return new Statistics(0);
+	}
+
+	@Override
+	public <E> Set<E> getDistinctFieldValues(final FieldDescriptor<Project, E> fieldDescriptor) {
+		return Collections.emptySet();
+	}
+}
```
[the remark this answers is not in the document]
Comment (tomvoelker) [reply bibsonomy-rest-api-v2/src/main/java/org/bibsonomy/api/search/NoOpProjectSearch.java]: Resolved: Converted to Kotlin in commit 8ac68e24da.

File: bibsonomy-rest-api-v2/src/main/java/org/bibsonomy/api/search/NoOpGroupSearch.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+/**
+ * Minimal no-op implementation to satisfy legacy beans when search is disabled.
+ */
+public class NoOpGroupSearch implements GroupSearch {
+
+	@Override
+	public List<Group> getGroups(final User loggedinUser, final GroupQuery query) {
+		return Collections.emptyList();
+	}
```
[the remark this answers is not in the document]
Comment (tomvoelker) [reply bibsonomy-rest-api-v2/src/main/java/org/bibsonomy/api/search/NoOpGroupSearch.java]: Declining: These are trivial no-op stubs with no meaningful logic to test. The project guidelines emphasize "integration tests (preferred)" and avoiding exhaustive test coverage. Testing that `emptyList()` returns an empty list adds no value and would be over-engineering.

File: bibsonomy-rest-api-v2/src/main/kotlin/org/bibsonomy/api/config/SecurityConfig.kt [hunk 5]
```diff
@@ -46,12 +48,18 @@ class SecurityConfig(
             .sessionManagement { it.sessionCreationPolicy(SessionCreationPolicy.STATELESS) }
             .authenticationManager(authenticationManager)
             .authorizeHttpRequests { authorize ->
+                val docMatchers = if (permitDocs) {
+                    arrayOf("/api/v2/api-docs/**", "/api/v2/swagger-ui.html", "/api/v2/swagger-ui/**")
+                } else {
+                    emptyArray()
+                }
```
[the remark this answers is not in the document]
Comment (tomvoelker) [reply bibsonomy-rest-api-v2/src/main/kotlin/org/bibsonomy/api/config/SecurityConfig.kt:55]: Declining: The current `if/else` form is explicit and readable. Using `takeIf` or `let` chains for simple conditionals can obscure intent. Preferring clarity over brevity here.

File: bibsonomy-webapp-v2/eslint.config.js [hunk 6]
```diff
@@ -70,12 +70,14 @@ export default [
       ],
       'vue/html-comment-content-spacing': ['error', 'always'],
       'vue/padding-line-between-blocks': ['error', 'always'],
+      'vue/attributes-order': 'off',
+      'no-undef': 'off',
```
[the remark this answers is not in the document]
Comment (tomvoelker) [reply bibsonomy-webapp-v2/eslint.config.js:74]: Declining: In Vue 3 with TypeScript and `<script setup>`, TypeScript's type checking fully covers undefined variable detection. The `no-undef` rule is known to produce false positives with Vue 3's compiler macros (`defineProps`, `defineEmits`, etc.) and auto-imports. Disabling it for Vue files is standard practice when TypeScript is in use.

File: bibsonomy-webapp-v2/eslint.config.js [hunk 7]
```diff
@@ -107,6 +109,8 @@ export default [
           prefer: 'type-imports',
         },
       ],
+      'prefer-promise-reject-errors': 'off',
+      'no-undef': 'off',
```
[the remark this answers is not in the document]
Comment (tomvoelker) [reply bibsonomy-webapp-v2/eslint.config.js:113]: Declining: Same reasoning as above - TypeScript handles undefined variable detection in `.ts` files. The `no-undef` rule is redundant when TypeScript strict mode is enabled and can cause false positives with type-only imports.

File: bibsonomy-rest-api-v2/src/main/kotlin/org/bibsonomy/api/config/ApiCorsConfig.kt [hunk 8]
```diff
@@ -11,11 +11,26 @@ class ApiCorsConfig {
...
+        val extraOrigins = System.getenv("BIBSONOMY_CORS_ALLOWED_ORIGINS")
+            ?.split(",")
+            ?.map { it.trim() }
+            ?.filter { it.isNotEmpty() }
+            ?: emptyList()
+        val originPatterns = System.getenv("BIBSONOMY_CORS_ALLOWED_ORIGIN_PATTERNS")
+            ?.split(",")
+            ?.map { it.trim() }
+            ?.filter { it.isNotEmpty() }
+            ?: emptyList()
```
[the remark this answers is not in the document]
Comment (tomvoelker) [reply bibsonomy-rest-api-v2/src/main/kotlin/org/bibsonomy/api/config/ApiCorsConfig.kt]: Resolved: Refactored to use Spring's @Value annotation in commit c990da00ce.

File: bibsonomy-webapp-v2/src/components/layout/Sidebar.vue [hunk 9]
```diff
@@ -87,12 +74,8 @@ const tagSizes = computed(() => {
...
     <!-- Recent Activity (placeholder) -->
     <div class="pt-4 border-t border-gray-300">
-      <h3 class="text-sm font-bold text-gray-800 mb-3">
-        Recent Activity
-      </h3>
-      <p class="text-sm text-gray-600">
-        Recent posts will appear here...
-      </p>
+      <h3 class="text-sm font-bold text-gray-800 mb-3">Recent Activity</h3>
+      <p class="text-sm text-gray-600">Recent posts will appear here...</p>
```
[the remark this answers is not in the document]
Comment (tomvoelker) [reply bibsonomy-webapp-v2/src/components/layout/Sidebar.vue]: Resolved: Added i18n keys for activity.recent and activity.placeholder in commit 1b3fd563b5.

File: bibsonomy-database/src/test/resources/database/insert-test-data.sql [hunk 10]
```diff
@@ -500,6 +500,9 @@ INSERT INTO `tags` (`tag_id`, `tag_name`, `tag_stem`, `tag_ctr`, `tag_ctr_public
 (21052615,'andere','',1,0,0),
 (21052616,'person','',1,0,0);
 
+-- Ensure popular tags are visible in minimal preview environments.
+UPDATE `tags` SET `show_tag` = 1;
+
```
[the remark this answers is not in the document]
Comment (tomvoelker) [reply bibsonomy-database/src/test/resources/database/insert-test-data.sql:507]: Resolved: Added documentation comments explaining the intentional unbounded UPDATE in commit be130cabe0.

File: bibsonomy-database/src/test/resources/database/insert-test-data.sql [hunk 11]
```diff
@@ -500,6 +500,16 @@ INSERT INTO `tags` (`tag_id`, `tag_name`, `tag_stem`, `tag_ctr`, `tag_ctr_public
...
+-- Seed popular_tags so /api/v2/tags returns data for ALL content type.
+DELETE FROM `popular_tags`;
+INSERT INTO `popular_tags` (`tag_lower`, `tag_ctr`, `content_type`, `popular_days`)
+  SELECT LOWER(`tag_name`), `tag_ctr_public`, 0, 0
+    FROM `tags`
+   WHERE `tag_ctr_public` > 0;
+
```
[the remark this answers is not in the document]
Comment (tomvoelker) [reply bibsonomy-database/src/test/resources/database/insert-test-data.sql:516]: Resolved: Added documentation comments explaining the intentional DELETE for idempotent test data seeding in commit be130cabe0.

File: bibsonomy-rest-api-v2/src/main/kotlin/org/bibsonomy/api/config/SecurityConfig.kt [hunk 12]
```diff
@@ -25,7 +26,8 @@ import org.springframework.security.config.Customizer
 @EnableWebSecurity
 class SecurityConfig(
     private val legacyBasicAuthenticationProvider: LegacyBasicAuthenticationProvider,
-    private val legacyAuthenticationEntryPoint: AuthenticationEntryPoint
+    private val legacyAuthenticationEntryPoint: AuthenticationEntryPoint,
+    @Value("\${security.permitDocs:false}") private val permitDocs: Boolean
```
[the remark this answers is not in the document]
Comment (tomvoelker) [reply bibsonomy-rest-api-v2/src/main/kotlin/org/bibsonomy/api/config/SecurityConfig.kt:30]: The permitDocs flag is a simple boolean for enabling Swagger UI in non-production environments. It's tested implicitly through integration tests that verify the API documentation endpoints are accessible. Adding dedicated unit tests for a simple boolean configuration property would be over-engineering.

File: bibsonomy-rest-api-v2/src/main/kotlin/org/bibsonomy/api/config/ApiCorsConfig.kt [hunk 13]
```diff
@@ -11,11 +11,26 @@ class ApiCorsConfig {
...
-                "http://localhost:5173",
-                "http://localhost:4173"
-            )
+            allowedOrigins = defaultOrigins + extraOrigins
+            if (originPatterns.isNotEmpty()) {
+                allowedOriginPatterns = originPatterns
+            }
             allowedMethods = listOf("GET", "POST", "PUT", "PATCH", "DELETE", "OPTIONS")
             allowedHeaders = listOf("Authorization", "Content-Type", "Accept")
             allowCredentials = true
```
[the remark this answers is not in the document]
Comment (tomvoelker) [reply bibsonomy-rest-api-v2/src/main/kotlin/org/bibsonomy/api/config/ApiCorsConfig.kt:38]: The allowCredentials=true is intentional for cookie-based authentication. The CORS configuration uses specific origins (localhost ports) by default, not wildcards. Additional origin patterns are opt-in via configuration. This follows CORS security best practices.
